cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -249,21 +249,21 @@ private DatasetMeta getFromMds(DatasetId instance) throws Exception {
   void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
     NamespaceId namespace = ConversionHelpers.toNamespaceId(namespaceId);
     DatasetId datasetId = ConversionHelpers.toDatasetInstanceId(namespaceId, name);
-    Principal principal = authenticationContext.getPrincipal();
+    Principal requestingUser = authenticationContext.getPrincipal();
     String ownerPrincipal = props.getOwnerPrincipal();
 
     // need to enforce on the principal id if impersonation is involved
-    KerberosPrincipalId principalId = ownerPrincipal == null ? null : new KerberosPrincipalId(ownerPrincipal);
-    if (!namespace.equals(NamespaceId.SYSTEM) && principalId == null) {
+    KerberosPrincipalId effectiveOwner = ownerPrincipal == null ? null : new KerberosPrincipalId(ownerPrincipal);
+    if (!namespace.equals(NamespaceId.SYSTEM) && effectiveOwner == null) {
```

Comment:
Is this also the same logic as the app check? If so we can rename to refactored method as `getEffectiveOwner()`.